src/main/java/io/lettuce/core/cluster/RedisClusterClient.java

```diff
@@ -936,6 +938,30 @@ protected Partitions loadPartitions() {
         return get(loadPartitionsAsync(), Function.identity());
     }
 
+    public void reauthenticateAllConnections(RedisCredentials creds) {
+        forEachClusterConnection(input -> {
+            input.reauthenticate(creds);
+        });
+
+        forEachClusterPubSubConnection(input -> {
+            input.reauthenticate(creds);
+        });
+    }
+    
+    public void reauthConnections() {
+        RedisCredentialsProvider credentialsProvider = getFirstUri().getCredentialsProvider();
+        if (credentialsProvider instanceof RedisCredentialsProvider.ImmediateRedisCredentialsProvider) {
```

Comment:
same (is this required, shouldn't there be an `else` clause?

Reply:
the same answer as above